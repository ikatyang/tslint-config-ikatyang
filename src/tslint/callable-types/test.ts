// good

type good1 = (x: any) => any;

// bad

type bad1 = {
  (x: any): any;
}

interface bad2 {
  (x: any): any;
}
