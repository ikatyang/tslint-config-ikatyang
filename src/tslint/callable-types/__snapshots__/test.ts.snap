// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lint 1`] = `
// good

type good1 = (x: any) => any;

// bad

type bad1 = {
  (x: any): any;
  ~~~~~~~~~~~~~~ [Type literal has only a call signature â€” use \`(x: any) => any\` instead.]
}

interface bad2 {
  (x: any): any;
  ~~~~~~~~~~~~~~ [Interface has only a call signature â€” use \`type bad2 = (x: any) => any;\` instead.]
}

`;
